src/main/java/pl/cleankod/exchange/provider/CurrencyConversionNbpService.java

```diff
@@ -16,11 +17,13 @@ public CurrencyConversionNbpService(ExchangeRatesNbpClient exchangeRatesNbpClien
         this.exchangeRatesNbpClient = exchangeRatesNbpClient;
     }
 
+
+    //we should correct the code, because if account currency != PLN then, conversion is wrong, because NBP provides rates for PLN
     @Override
     public Money convert(Money money, Currency targetCurrency) {
-        RateWrapper rateWrapper = exchangeRatesNbpClient.fetch("A", targetCurrency.getCurrencyCode());
+        RateWrapper rateWrapper = exchangeRatesNbpClient.fetch("A", targetCurrency.name());
         BigDecimal midRate = rateWrapper.rates().get(0).mid();
-        BigDecimal calculatedRate = money.amount().divide(midRate, RoundingMode.HALF_UP);
-        return new Money(calculatedRate, targetCurrency);
+        BigDecimal calculatedRate = money.amount().divide(midRate,10, RoundingMode.HALF_EVEN);
```

Comment:
can you elaborate. I know it could be done better, but real rounding happens line below. this is just getting pretty much ok result that needs some corrections